lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -230,6 +230,72 @@ def add_merchant_description(post, options)
         merchant[:locality] = options[:merchant_descriptor_locality] if options[:merchant_descriptor_locality]
       end
 
+      def add_stored_credentials(post, payment, options)
+        stored_credential = options[:stored_credential]
+        if stored_credential
+          options = stored_credential_options(stored_credential, options.fetch(:reason_code, ''))
+          post[:processingInformation][:commerceIndicator] = options.fetch(:transaction_type, 'internet')
+          stored_credential[:initial_transaction] ? initial_transaction(post, options) : subsequent_transaction(post, options)
+        end
+      end
+
+      def stored_credential_options(options, reason_code)
+        transaction_type = options[:reason_type]
+        transaction_type = 'install' if transaction_type == 'installment'
+        initiator = options[:initiator] if  options[:initiator]
+        initiator = 'customer' if initiator == 'cardholder'
+        stored_on_file = options[:reason_type] == 'recurring'
+        options.merge({
+          transaction_type: transaction_type,
+          initiator: initiator,
+          reason_code: reason_code,
+          stored_on_file: stored_on_file
+        })
+      end
+
+      def add_processing_information(initiator, merchant_initiated_transaction_hash = {})
+        {
+          authorizationOptions: {
+            initiator: {
+              type: initiator,
+              merchantInitiatedTransaction: merchant_initiated_transaction_hash
+            }
+          }
+        }.compact
+      end
+
+      def initial_transaction(post, options)
+        processing_information = add_processing_information(options[:initiator], {
+          reason: options[:reason_code]
+        })
+
+        post[:processingInformation].merge!(processing_information)
+      end
+
+      def subsequent_transaction(post, options)
+        processor_information = add_processor_information(post, options)
+        network_transaction_id = processor_information ? processor_information[:transactionId] : ''
```

Comment:
Seems like this can be simplified a lot? Or I am missing something
```suggestion
        network_transaction_id = options[:network_transaction_id] || options.dig(:stored_credential, :network_transaction_id) || ''
```

Reply:
maybe yes, but I'm using the `add_processor_information` function to add the `processorInformation` object
```
  post[:processorInformation] = {
          transactionId: ntid
  }
```

Reply:
Updated, finally `transactionId` is not necessary